osu.Game/Overlays/Rankings/Tables/RankingsTable.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using System.Collections.Generic;
+using System.Linq;
+using osu.Framework.Allocation;
+using osu.Framework.Extensions;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using osu.Framework.Extensions.IEnumerableExtensions;
+
+namespace osu.Game.Overlays.Rankings.Tables
+{
+    public abstract class RankingsTable<TModel> : TableContainer
+    {
+        protected const int TEXT_SIZE = 14;
+        private const float horizontal_inset = 20;
+        private const float row_height = 25;
+        private const int items_per_page = 50;
+
+        private readonly int page;
+        private readonly FillFlowContainer backgroundFlow;
+
+        public IReadOnlyList<TModel> Rankings
+        {
+            set
+            {
+                Content = null;
+                backgroundFlow.Clear();
+
+                if (value?.Any() != true)
+                    return;
+
+                value.ForEach(_ => backgroundFlow.Add(new TableRowBackground()));
+
+                Columns = CreateHeaders();
+                Content = value.Select((s, i) => CreateContent(page * items_per_page - (items_per_page - i), s)).ToArray().ToRectangular();
```

Comment:
```suggestion
                Content = value.Select((s, i) => CreateContent((page - 1) * items_per_page + i, s)).ToArray().ToRectangular();
```

i've never seen such convoluted number logic